### PR TITLE
Add app exit on "Go back"

### PR DESCRIPTION
**Pull request by dmitrylyzo**
This PR is a continuation of #702 (requires "Back" handling from it).

Add app exit if no more "Back".

On Tizen, user will be asked about exit.

On WebOS, home menu will be shown. Requires disabling "Back" handling by WebOS in app. There seems to be no harm to an actual WebOS app, as it intercepts `keydown` event for "Back" button.

### CODE REVIEW

**Comment by JustAMan** [comment]
Blocked by https://github.com/jellyfin/jellyfin-web/pull/702 not merged yet

**Comment by dmitrylyzo** [comment]
To make navigation in "iframed" webos app work like that
jellyfin-page-nav2-2

I need to mark `login.html` with `type: 'login'` in `routes.js` and treat this as no "go back".

<details>
<summary>Changes</summary>

```diff
diff -ruN jellyfin-web.master12/components/appRouter.js jellyfin-web.master12~key/components/appRouter.js
--- jellyfin-web.master12/components/appRouter.js	2020-01-19 20:41:13.253551115 +0300
+++ jellyfin-web.master12~key/components/appRouter.js	2020-01-21 22:56:15.002371729 +0300
@@ -545,13 +545,20 @@
         page.back();
     }
 
+    /**
+     * Pages of "no return" (Go back).
+     */
+    var startPages = ['home', 'login'];
+
     function canGoBack() {
         var curr = current();
         if (!curr) {
             return false;
         }
 
-        if (curr.type === 'home') {
+        if (!document.querySelector('.dialogContainer') && startPages.indexOf(curr.type) !== -1) {
             return false;
         }
         return page.canGoBack();
diff -ruN jellyfin-web.master12/scripts/routes.js jellyfin-web.master12~key/scripts/routes.js
--- jellyfin-web.master12/scripts/routes.js	2020-01-19 20:41:13.293550788 +0300
+++ jellyfin-web.master12~key/scripts/routes.js	2020-01-21 23:00:31.755042698 +0300
@@ -226,7 +226,8 @@
         autoFocus: false,
         anonymous: true,
         startup: true,
-        controller: "auth/login"
+        controller: "auth/login",
+        type: "login"
     });
     defineRoute({
         path: "/metadataimages.html",
```
</details>

Any objections?

**Comment by JustAMan** [comment]
> Any objections?

Looks fine to me

**Comment by dkanada** [comment]
One quick note, the library we use here has custom changes that should be removed eventually, which might break this.

**Comment by dmitrylyzo** [comment]
> One quick note, the library we use here has custom changes that should be removed eventually, which might break this.

Which one?

**Comment by dkanada** [comment]
page.js

File: src/components/appRouter.js [hunk 1]
```diff
@@ -545,13 +545,18 @@ define(['loading', 'globalize', 'events', 'viewManager', 'layoutManager', 'skinM
         page.back();
     }
 
+    /**
+     * Pages of "no return" (Go back).
```
Comment (JustAMan) [inline src/components/appRouter.js]:
```suggestion
     * Pages of "no return" (when "Go back" should behave differently, probably quitting the application).
```

**Pull request merged**